python_modules/dagster/dagster/_grpc/server.py

```diff
@@ -429,48 +427,43 @@ def ExternalPartitionSetExecutionParams(self, request, _context):
         yield from self._split_serialized_data_into_chunk_events(serialized_data)
 
     def ExternalPartitionConfig(self, request, _context):
-        args = deserialize_json_to_dagster_namedtuple(request.serialized_partition_args)
-
-        check.inst_param(args, "args", PartitionArgs)
-
-        recon_repo = self._recon_repository_from_origin(args.repository_origin)
+        args = deserialize_as(request.serialized_partition_args, PartitionArgs)
 
         return api_pb2.ExternalPartitionConfigReply(
             serialized_external_partition_config_or_external_partition_execution_error=serialize_dagster_namedtuple(
-                get_partition_config(recon_repo, args.partition_set_name, args.partition_name)
+                get_partition_config(
+                    self._get_repo_for_origin(args.repository_origin),
+                    args.partition_set_name,
+                    args.partition_name,
+                )
             )
         )
 
     def ExternalPartitionTags(self, request, _context):
-        partition_args = deserialize_json_to_dagster_namedtuple(request.serialized_partition_args)
-
-        check.inst_param(partition_args, "partition_args", PartitionArgs)
-
-        recon_repo = self._recon_repository_from_origin(partition_args.repository_origin)
+        partition_args = deserialize_as(request.serialized_partition_args, PartitionArgs)
 
         return api_pb2.ExternalPartitionTagsReply(
             serialized_external_partition_tags_or_external_partition_execution_error=serialize_dagster_namedtuple(
                 get_partition_tags(
-                    recon_repo, partition_args.partition_set_name, partition_args.partition_name
+                    self._get_repo_for_origin(partition_args.repository_origin),
+                    partition_args.partition_set_name,
+                    partition_args.partition_name,
                 )
             )
         )
 
     def ExternalPipelineSubsetSnapshot(self, request, _context):
-        pipeline_subset_snapshot_args = deserialize_json_to_dagster_namedtuple(
-            request.serialized_pipeline_subset_snapshot_args
-        )
-
-        check.inst_param(
-            pipeline_subset_snapshot_args,
-            "pipeline_subset_snapshot_args",
+        pipeline_subset_snapshot_args = deserialize_as(
+            request.serialized_pipeline_subset_snapshot_args,
             PipelineSubsetSnapshotArgs,
         )
 
         return api_pb2.ExternalPipelineSubsetSnapshotReply(
             serialized_external_pipeline_subset_result=serialize_dagster_namedtuple(
                 get_external_pipeline_subset_result(
-                    self._recon_pipeline_from_origin(pipeline_subset_snapshot_args.pipeline_origin),
+                    self._get_recon_pipeline_for_origin(
+                        pipeline_subset_snapshot_args.pipeline_origin
+                    ),
```

Comment:
so close to removing reconstructableXXX from the server altogether.. but I peeked into get_external_pipeline_subset_result and it does look dicey. Maybe not impossible though?

Reply:
k i did this - re-requesting review